File: algorithm-leetcode-jvm/src/test/java/org/muxun/algorithm/leetcode/java/array/LeetCode2529Test.java
package org.muxun.algorithm.leetcode.java.array;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

/**
 * @author 慕勋
 * @created 2024-04-09
 */
@DisplayName("正整数和负整数的最大计数测试用例")
public class LeetCode2529Test {

    @Test
    public void testOne() {
        int[] nums = {-2, -1, -1, 1, 2, 3};
        int expected = 3;
        int res = new LeetCode2529().maximumCount(nums);
        Assertions.assertEquals(expected, res);
    }

    @Test
    public void testTwo() {
        int[] nums = {-3, -2, -1, 0, 0, 1, 2};
        int expected = 3;
        int res = new LeetCode2529().maximumCount(nums);
        Assertions.assertEquals(expected, res);
    }

    @Test
    public void testThree() {
        int[] nums = {5, 20, 66, 1314};
        int expected = 4;
        int res = new LeetCode2529().maximumCount(nums);
        Assertions.assertEquals(expected, res);
    }
}

File: algorithm-leetcode-jvm/src/test/kotlin/org/muxun/algorithm/leetcode/kotlin/array/LeetCode2529Test.kt
package org.muxun.algorithm.leetcode.kotlin.array

import kotlin.test.Test
import kotlin.test.assertEquals

class LeetCode2529Test {

    @Test
    fun testOne() {
        val nums = intArrayOf(-2, -1, -1, 1, 2, 3)
        val expected = 3
        val res = LeetCode2529().maximumCount(nums)
        assertEquals(expected, res)
    }

    @Test
    fun testTwo() {
        val nums = intArrayOf(-3, -2, -1, 0, 0, 1, 2)
        val expected = 3
        val res = LeetCode2529().maximumCount(nums)
        assertEquals(expected, res)
    }

    @Test
    fun testThree() {
        val nums = intArrayOf(5, 20, 66, 1314)
        val expected = 4
        val res = LeetCode2529().maximumCount(nums)
        assertEquals(expected, res)
    }
}